### PR TITLE
Create Component-Types controller

### PR DESCRIPTION
**Pull request by hkhattabii**
Able the iesi web appplication to fetch the component-types

### CODE REVIEW

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/ComponentTypeDtoService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        this.componentTypeParameterDtoService = componentTypeParameterDtoService;
+    }
+
+    public ComponentTypeDto convertToDto(ComponentType componentType, String name) {
+        return new ComponentTypeDto(name, componentType.getDescription(),
+                componentType.getParameters().entrySet().stream()
+                        .map(entry -> componentTypeParameterDtoService.convertToDto(entry.getValue(), entry.getKey()))
+                        .collect(Collectors.toList()));
+    }
+}
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/ComponentTypeDtoService.java:24]: can't this be put in a ResourceAssembler?

File: core/java/iesi-rest-without-microservices/src/main/resources/application-iesi-connection-types.yml [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+            encrypted: false
+          port:
+            description: Set the specific port for the host. In case the default port (80 for http, 443 for HTTPS) this parameter can be omitted
+            type: number
+            mandatory: false
+            encrypted: false
+          baseUrl:
+            description: all endpoints hosted at the host start with this specific endpoint
+            type: string
+            mandatory: true
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/resources/application-iesi-connection-types.yml:20]: this is not mandatory

**Review by BerrevoetsRobbe — CHANGES_REQUESTED** [review]

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeParameterService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    public ComponentTypeParameterDto convertToDto(ComponentTypeParameter componentTypeParameter, String name) {
+        return new ComponentTypeParameterDto(
+                name,
+                componentTypeParameter.getDescription(),
+                componentTypeParameter.getType(),
+                componentTypeParameter.isMandatory(),
+                componentTypeParameter.isEncrypted()
+        );
+    }
+}
```
Comment (hkhattabii) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeParameterService.java:18]: It is not possible to add resource assembler support for the ComponentTypeParameterDto.
The ComponentTypeParameter doesn't have the name getter, so I had to pass the nam as an argument.
Also,  the RepresentationModelAssemblerSupport doesn't provide the methods ToModel with a second argument.

So I let the service to convert the ComponentTypeParameter to Dto

File: core/java/iesi-core/src/main/java/io/metadew/iesi/common/configuration/metadata/componenttypes/MetadataComponentTypesConfiguration.java [hunk 4]
```diff
@@ -4,6 +4,7 @@
 import io.metadew.iesi.common.configuration.Configuration;
 import io.metadew.iesi.common.configuration.metadata.MetadataConfiguration;
 import io.metadew.iesi.metadata.definition.component.ComponentType;
+import io.metadew.iesi.metadata.definition.connection.ConnectionType;
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/common/configuration/metadata/componenttypes/MetadataComponentTypesConfiguration.java:7]: unused import?

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeDtoResourceAssembler.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    private ComponentTypeDto convertToDto(ComponentType componentType) {
+        log.info(componentType);
+        return new ComponentTypeDto(
+                componentType.getName(),
+                componentType.getDescription(),
+                componentType.getParameters().entrySet().stream()
+                        .map(entry -> componentTypeParameterService.convertToDto(entry.getValue(), entry.getKey()))
+                        .collect(Collectors.toList())
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeDtoResourceAssembler.java:33]: Move logic of core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeParameterService.java to here and remove core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeParameterService.java

**Review by BerrevoetsRobbe — CHANGES_REQUESTED** [review]

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeDtoResourceAssembler.java (hunk 5 above)
Comment (BerrevoetsRobbe) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/componentTypes/dto/ComponentTypeDtoResourceAssembler.java:33]: bump

**Review by BerrevoetsRobbe — APPROVED** [review]

**Pull request merged**